app/src/main/java/com/jalasoft/routesapp/ui/auth/registerUser/view/RegisterUserFragment.kt

```diff
@@ -35,17 +50,41 @@ class RegisterUserFragment : Fragment() {
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? {
+    ): View {
         _binding = FragmentRegisterUserBinding.inflate(inflater, container, false)
         return binding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        viewModel.context = context
+        googleConfiguration()
+        facebookConfiguration()
+        buttonActions()
+    }
 
+    @Deprecated("Deprecated in Java")
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        callbackManager.onActivityResult(requestCode, resultCode, data)
+        super.onActivityResult(requestCode, resultCode, data)
+    }
+
+    private fun buttonActions() {
         binding.btnRegRegister.setOnClickListener {
             addUser()
         }
+
+        binding.ibRegGoogle.setOnClickListener {
+            signInGoogle()
+        }
+
+        binding.btnRegLogin.setOnClickListener {
+            findNavController().navigateUp()
+        }
+
+        binding.ibRegFacebook.setOnClickListener {
+            loginManager.logIn(this, listOf("email", "public_profile"))
```

Comment:
reuse the constant array here

Reply:
Also, I don't know why would you need a second parameter here if there´s always the same value in both calls, better use the parameter inside the login method.

Reply:
The login method comes from Facebook SDK, I can't change it